[bug] `update-rules` fails for packages created with `Release Automation`

**Describe the bug**
`update-rules` fails for packages created with `Release Automation`.

Sorry, I did not notice this issue because the rules folder was up to date when I was testing :(
This issue occurs when there is an update to `hayabusa_rules` repo after the Release Automation Actions are executed.

**Step to Reproduce**
1. Download package from https://github.com/Yamato-Security/hayabusa/actions/runs/11233651311
2. ./hayabusa update-rules

**Actual behavior**
```
% ./hayabusa-2.18.0-mac-aarch64 update-rules -q
Start time: 2024/10/09 09:14

[ERROR] Failed to update rules. Error { code: -1, klass: 9, message: "object not found - no match for id (3c2a557667c452d01ed9c869be0d86ee669cb78b)" }
```

**Expected behavior**
Successfully updated rules 

**Environment**
- OS: macOS 15.0.1
- hayabusa version: 2.18.0-dev

**Additional context**
I have been deleting unused files under the rules folder in the actions, so that may be the cause ... 🤔

In `Integration-test`, `cargo build --release` binaries are used. However, it does not use the `release package zip`, so it cannot detect this issue. It might be better to also have an `Integration-test` that uses the `release package zip`...

>[ERROR] Failed to update rules. Error { code: -1, klass: 9, message: "object not found - no match for id (3c2a557667c452d01ed9c869be0d86ee669cb78b)" }

The above message matches the following commit hash from this morning
https://github.com/Yamato-Security/hayabusa-rules/commit/3c2a557667c452d01ed9c869be0d86ee669cb78b

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -39,12 +39,12 @@ jobs:
           }
           - {
             os: "ubuntu-24.04",
-            target: "x86_64-unknown-linux-musl",
+            target: "aarch64-unknown-linux-gnu",
             cross: true,
           }
           - {
             os: "ubuntu-22.04",
-            target: "aarch64-unknown-linux-gnu",
+            target: "x86_64-unknown-linux-musl",
             cross: true,
           }
           - {

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -75,6 +75,7 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           repository: Yamato-Security/hayabusa-rules
+          fetch-depth: 0
           path: hayabusa-rules
 
       - name: Set up Rust toolchain

